java/semantickernel-api/src/main/java/com/microsoft/semantickernel/services/OrderedAIServiceSelector.java

```diff
@@ -0,0 +1,191 @@
+package com.microsoft.semantickernel.services;
+
+import com.microsoft.semantickernel.AIService;
+import com.microsoft.semantickernel.Verify;
+import com.microsoft.semantickernel.chatcompletion.ChatCompletionService;
+import com.microsoft.semantickernel.orchestration.KernelFunction;
+import com.microsoft.semantickernel.orchestration.PromptExecutionSettings;
+import com.microsoft.semantickernel.orchestration.contextvariables.KernelArguments;
+import com.microsoft.semantickernel.textcompletion.TextGenerationService;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OrderedAIServiceSelector extends ParentAIServiceSelector {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrderedAIServiceSelector.class);
+
+    public OrderedAIServiceSelector() {
+        super(new HashMap<>());
+    }
+
+    public OrderedAIServiceSelector(Map<Class<? extends AIService>, AIService> services) {
+        super(services);
+    }
+
+    @Nullable
+    public AIServiceSelection trySelectAIService(
```

Comment:
I think all of the fall through cases should be split out into their own methods so that the logic here could be  iterating over a stream of these select methods and doing a 'findFirst' kinda thing. This would make it easier to add services in the future.